### PR TITLE
Fix asset-url("/path", true)

### PR DESCRIPTION
**Pull request by glebm**
Same as PR #1724, but rebased with respect to the latest stable.

### CODE REVIEW

File: core/lib/compass/core/sass_extensions/functions/urls.rb [hunk 1]
```diff
@@ -241,12 +228,8 @@ def generated_image_url(path, cache_buster = bool(false))
   # Emits a path, taking off any leading "./"
   def clean_path(url)
     url = url.to_s
-    url = url[0..1] == "./" ? url[2..-1] : url
-  end
-
-  # Emits a url, taking off any leading "./"
-  def clean_url(url)
-    unquoted_string("url('#{clean_path(url)}')")
+    url = url[2..-1] if url.start_with?('./')
```
Comment (glebm) [inline core/lib/compass/core/sass_extensions/functions/urls.rb:231]: Why? Single quoted strings indicate simple non-interpolated strings.

Comment (Arcovion) [inline core/lib/compass/core/sass_extensions/functions/urls.rb:231]: Yea, you should always prefer single quoted strings. Also, line length 80? Really?
The styleguide this bot uses [is here](https://github.com/thoughtbot/guides/tree/master/style) - it's all very disagreeable, no vertical alignment, acronyms without capital letters, avoid conditional modifiers,  avoid bang methods?! It's not rubyish.

Suggest switching to RuboCop which is a more customisable, relaxed analyser that won't litter with comments and can automatically fix code by itself.

Comment (chriseppstein) [inline core/lib/compass/core/sass_extensions/functions/urls.rb:231]: We're just testing out this tool. We use rubocop on Sass and I liked it. But getting the code to comply is a TON of work.

Comment (Anahkiasen) [inline core/lib/compass/core/sass_extensions/functions/urls.rb:231]: @chriseppstein What about [Scrutinizer](https://scrutinizer-ci.com/) ? It aggregates various CI tools for PHP, Ruby, JS, and Python. It's quite powerful. Supports pull requests statuses and stuff.

**Comment by glebm** [comment]
@chriseppstein Does anything else need to be done on my side for this to be merged?

**Comment by chriseppstein** [comment]
I just re-wrote all our url handling. Grab compass 1.1.0.alpha.0 and let me know if it's working correctly for you.

**Comment by glebm** [comment]
I've opened a separate PR with just the test that was previously failing but now passes with the current Compass master. https://github.com/Compass/compass/pull/1812

**Pull request closed without merging**